### ISSUE TITLE
Repo to be removed

### ISSUE THREAD

**Issue by hawesie** [strands-project/autonomous_patrolling#128]
For the release process (https://github.com/strands-project/strands_management/wiki/software-distribution) this repo will be merged into https://github.com/strands-project/strands_navigation/ and restructured.

**Comment by hawesie** [comment]
@marc-hanheide we discussed having a strands-attic org to keep old code. Shall we do this? I think this repo would be a good candidate for that once I have copied the relevant code.

**Comment by marc-hanheide** [comment]
:+1

**Comment by Jailander** [comment]
the  topological_patroller is very data gathering specific I don't know if it should be moved to a release repo like strands_navigation or to a more specific repo like Linda?

**Comment by hawesie** [comment]
The topological patroller is pretty complex iirc and we could probably put a simpler version in strands_navigation if we need it.

**Comment by Jailander** [comment]
there is a simpler version now, I can clean it up a little and put it in the strands navigation repository in that case

**Comment by hawesie** [comment]
Ok.

**Comment by marc-hanheide** [comment]
so... what's the update on this? I would even suggest, @Jailander keeps whatever code is still remaining as attic in his own github acount and we delete this one ASAP (i.e. as soon as everything that needs moving is moved).

**Comment by Jailander** [comment]
From my part it can be removed I will upload the patrol server to strands_navigation today and the rest can go, I also have a copy of everything in my repo

**Comment by bfalacerda** [comment]
Just to remind you that the gamepad based map saver and waypoint recorder packages are also here. They are pretty important to set up the system

**Comment by Jailander** [comment]
hmm yes, that is true, maybe we can move them to strands_navigation temporarily?

**Comment by marc-hanheide** [comment]
Why temporarily? I want something sustainable

**Comment by Jailander** [comment]
well the map_saver should be there or somewhere else permanently and the waypoint recorder should be modified to record nodes instead I just didn't think about it until now, and now I want to do it I think it will make the topological map building a lot easier

**Comment by marc-hanheide** [comment]
Indeed. So just include it in strands_navigation properly I'd say.

**Comment by Jailander** [comment]
Ok, but it will take a while then

**Comment by marc-hanheide** [comment]
why? ;-)

**Comment by Jailander** [comment]
because I wanted to merge all my branch and I am doing testing to make sure I don't break anything else

**Comment by bfalacerda** [comment]
There might also be an issue with the dependency to scitos_teleop, as we don't want strands_navigation to depend on scitos_apps. I see that @cdondrup did something to look for scitos_teleop and compile with an extra flag when it is found, is that enough?

**Comment by cdondrup** [comment]
That is to use the buttons on the joypad when running the waypoint or map recorder. I have my own message for the buttons I use. I could rewrite it so it uses the `sensor_msgs::Joy` messages instead of those. If you want joypad support at all.

**Comment by Jailander** [comment]
Yes that was my plan, I was thinking about moving the node saver to `strands_navigation` doing that and maybe the map saver can go to `scitos_apps` where there is no need to change anything. @marc-hanheide is that ok?

**Comment by cdondrup** [comment]
The map_saver is not really robot specific and should go to `strands_apps` then. But getting rid of the `scitos_teleop` dependency is not hard.

**Comment by Jailander** [comment]
but in `strands_apps` there are inter repository dependencies unlike `scitos_apps` what is the difference between these two repos I think the map_saver can be considered robot specific since ROS has other methods to save the map, anyway at some point we should start saving the maps in the mongodb

**Comment by cdondrup** [comment]
Robot specific means that it won't run on any other robot than ours due to the specific hardware or configuration. The `map_saver` just encapsulates the ROS `map_server` function to save maps to be able to use it via a joypad. Therefore it can run on any robot that uses ROS and should be in `strands_apps`.

The difference between the two repos is therefore that `scitos_apps` contains tools that only run on a SCITOS G5 due to hardware or configuration specifications and `strands_apps` contains tools that run on any ROS system.

Long story short, imho the `map_saver` should go into strands_apps or be deleted completely because it doesn't offer any functionality except for saving maps via a button press.

**Comment by Jailander** [comment]
Ok, to start with I will move the waypoint saver to `strands_navigation` and modify it to create nodes instead of waypoints using the `sensor_msgs::Joy` message and afterwards I'll see about the map saver since it shouldn't go in `strands_navigation` afterwards

**Comment by marc-hanheide** [comment]
just saying that conditional compilation doesn't help, because in the released packages it simply wouldn't be build then. So, only depending it on `sensor_msgs::Joy` is the sensible option, I'd say. Maybe the map_saver should be renamed to `map_save_joy_trigger` or something? Besides that, @cdondrup is 100% correct about the nature of the repos.

**Comment by Jailander** [comment]
Sorry maybe I expressed myself wrong, I wasn't discussing the nature of the repos, I was just saying that joystick configuration is robot dependant, in any case I am modifying both modules to use the `sensor_msgs::Joy` message and I will move `map_saver` to `joy_map_save` in the `strands_apps` I will create parameters for the triggers of each actions (node server). 

Btw the node saver will be an script in the topological navigation package, unless someone disagrees. It will all be ready today in the afternoon.

**Comment by hawesie** [comment]
> On 3 Nov 2014, at 14:04, Jaime Pulido Fentanes notifications@github.com wrote:
> 
> Btw the node saver will be an script in the topological navigation package, unless someone disagrees. It will all be ready today. 

Everything else of this nature is in topological_utils, so I suggest putting it there for consistency.=

**Comment by Jailander** [comment]
Ok no problem

**Comment by Jailander** [comment]
Hi, pull request #131 removes some deprecated packages can someone else check the remaining ones I am testing the new map saver nodes (metric and topological once its done) I'll create pull requests for them on `strands_apps` and `strands_navigation`
